test/elixir_bench_web/schema/schema_test.exs

```diff
@@ -79,30 +81,403 @@ defmodule ElixirBenchWeb.SchemaTest do
+          name: "insert_mysql/insert_plain"
+        ) {
+          name
+        }
+      """
+
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      error_message = Enum.at(resp["errors"], 0)
+
+      assert %{"message" => "not_found"} = error_message
+    end
+
+    test "return required fields message when missing fields", context do
+      required_fields = ["repoSlug", "name"]
+
+      query = """
+        benchmark {
+          name
+        }
+      """
+
+      resp_body = context.conn
+        |> execute_query(query)
+        |> Map.get(:resp_body)
+
+      Enum.each(required_fields, fn field ->
+        assert(
+          Regex.match?(~r/errors.*#{field}.*found null/, resp_body),
+          "Field '#{field}' not found in #{resp_body}"
+        )
+      end)
+    end
+  end
+
+  describe "measurement query" do
+    test "fetch measurement by id", context do
+      measurement = insert(:measurement)
+      job = measurement.job
+
+      query = """
+        measurement (id: #{measurement.id})
+        {
+          id
+          collectedAt
+          commit {
+            message
+            sha
+            url
+          }
+          environment {
+            cpu
+            cpuCount
+            elixirVersion
+            erlangVersion
+            memory
+          }
+          result {
+            average
+            ips
+            maximum
+            median
+            minimum
+            mode
+            sampleSize
+            stdDev
+            stdDevIps
+            stdDevRatio
+          }
+        }
+      """
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      json_data =
+        %{
+          "data" => %{
+            "measurement" => %{
+              "id" => "#{measurement.id}",
+              "collectedAt" => job.completed_at |> DateTime.to_iso8601,
+              "commit" => %{
+                "message" => "#{job.commit_message}",
+                "sha" => "#{job.commit_sha}",
+                "url" => "#{job.commit_url}"
+              },
+              "environment" => %{
+                "cpu" => "#{job.cpu}",
+                "cpuCount" => job.cpu_count,
+                "elixirVersion" => "#{job.elixir_version}",
+                "erlangVersion" => "#{job.erlang_version}",
+                "memory" => job.memory_mb
+              },
+              "result" => %{
+                "average" => measurement.average,
+                "ips" => measurement.ips,
+                "maximum" => measurement.maximum,
+                "median" => measurement.median,
+                "minimum" => measurement.minimum,
+                "mode" => measurement.mode,
+                "sampleSize" => measurement.sample_size,
+                "stdDev" => measurement.std_dev,
+                "stdDevIps" => measurement.std_dev_ips,
+                "stdDevRatio" => measurement.std_dev_ratio
```

Comment:
not sure if we can do this, but this seems rather repetitive... but our original keys are snake_case... so yeah maybe it has to be. For another case I made a simple convert utility that given the original struct adjusts keys etc but this should be fine I guess :)

Reply:
Looks like camelCase is the default JS format. [Asinthe docs](https://hexdocs.pm/absinthe/Absinthe.Adapter.html#content) explains about two adapters, one that does transformation and a ```Passthrough``` adapter, that relies on the structure given in the request. If you request CamelCase you receive CamelCase, if you request snake_case you receive snake_case. We can create an issue to change this later as it would need some other settings.